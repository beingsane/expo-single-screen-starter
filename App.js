import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { colors } from './src/api/constants';

// screens
import Menu from './src/screens/Menu';

// component(s)
import Touch from './src/components/Touch';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showLeftMenu: false,
      showRightMenu: false
    };

    this.menuOpen = this.menuOpen.bind(this);
    this.menuClose = this.menuClose.bind(this);
  }

  menuOpen(showMenu) {
    this.setState({
      showLeftMenu: showMenu === 'showLeftMenu',
      showRightMenu: showMenu === 'showRightMenu'
    });
  }

  menuClose() {
    this.setState({
      showLeftMenu: false,
      showRightMenu: false
    });
  }

  render() {
    const { showLeftMenu, showRightMenu } = this.state;

    return (
      <View style={styles.container}>
        <Text style={styles.text}>welcome!</Text>
        <Text style={styles.text}>
          this is a simple one page react native app
        </Text>

        <Touch
          style={[styles.block, styles.leftBlock]}
          text="left"
          onPress={() => this.menuOpen('showLeftMenu')}
        />
        <Touch
          style={[styles.block, styles.rightBlock]}
          text="right"
          onPress={() => this.menuOpen('showRightMenu')}
        />

        <Menu show={showLeftMenu} onClose={this.menuClose} />
        <Menu direction="right" show={showRightMenu} onClose={this.menuClose} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: colors.brandPrimary,
    flex: 1,
    justifyContent: 'center'
  },
  block: {
    alignItems: 'center',
    bottom: 16,
    height: 72,
    position: 'absolute',
    justifyContent: 'center',
    width: 72
  },
  leftBlock: {
    backgroundColor: colors.blue,
    left: 16
  },
  rightBlock: {
    backgroundColor: colors.brown,
    right: 16
  },
  text: {
    color: colors.white
  }
});
