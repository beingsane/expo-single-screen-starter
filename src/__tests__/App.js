import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../App';

it('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  // this too
  // https://medium.com/codeclan/testing-react-with-jest-and-enzyme-20505fec4675
  // use shallow
  // const loginComponent = shallow(<App />);
  // loginComponent.setState({ showLeftMenu: true });

  expect(tree).toMatchSnapshot();
});
