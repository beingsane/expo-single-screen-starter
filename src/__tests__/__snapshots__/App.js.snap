// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App :: renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1d2d3a",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    welcome!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    this is a simple one page react native app
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0197f6",
        "borderRadius": 4,
        "bottom": 16,
        "height": 64,
        "justifyContent": "center",
        "left": 16,
        "opacity": 1,
        "padding": 4,
        "position": "absolute",
        "width": 64,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      left menu
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0197f6",
        "borderRadius": 4,
        "bottom": 16,
        "height": 64,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 4,
        "position": "absolute",
        "right": 16,
        "width": 64,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      right menu
    </Text>
  </View>
</View>
`;
